### PR TITLE
Keep NaN last-window levels for NaN-tolerant estimators (#1196)

### LINKED ISSUE #1196: [FEATURE]: Add min_non_nan_last_window for multiseries folds (min_periods-like behavior)
### Describe the feature

In multiseries workflows, `dropna_from_series=True` currently drops a level from `series_last_window` if that level contains **any** NaN in the window (`dropna(axis=1, how="any")` in `_extract_data_folds_multiseries`).

I would like to propose a less strict option, similar in spirit to pandas/sktime-style minimum-valid-observation handling: keep a level if it has at least a configurable number of non-NaN values in the last window.

### Motivation

Current behavior is all-or-nothing:
- 1 NaN in the last window => whole level is dropped for that fold.

In practice, some use cases only need “enough” valid values, not perfect completeness.  
A configurable threshold would reduce unnecessary level drops while preserving safety.

### Proposed API

Add a new optional parameter for multiseries forecasters (and propagate it into `_extract_data_folds_multiseries`), for example:

- `min_non_nan_last_window: int | None = None`

Proposed behavior:
- If `dropna_from_series=False`: unchanged.
- If `dropna_from_series=True` and `min_non_nan_last_window is None`: keep current behavior (`how="any"`).
- If `dropna_from_series=True` and `min_non_nan_last_window=k`: drop level only when non-NaN count in the last window is `< k`.

Validation:
- Require `k >= 1`
- If `k` is greater than the fold last-window length, raise `ValueError`.

### Expected impact

- Backward compatible by default.
- Gives users controlled flexibility similar to `min_periods` logic.
- Useful for multiseries datasets with sparse/mildly missing tail values.

### References

- `_extract_data_folds_multiseries` TODO comment in `skforecast/model_selection/_utils.py`
- Related concept: minimum-valid-observation thresholds (e.g., pandas rolling `min_periods`)

### PR DESCRIPTION
**Pull request by vikas-prasad-cx**
## Summary
- Implements the maintainer-preferred design for #1196: when \`dropna_last_window=True\`, keep levels whose last window contains NaNs **only if** the estimator natively supports NaN inputs and the forecaster has **no differentiation**.
- Adds \`estimator_has_native_nan_support\` in \`skforecast.utils\`, reusing the module-based family detection from \`configure_estimator_categorical_features\` (lightgbm / catboost / xgboost / sklearn HistGradientBoosting) and unwrapping \`Pipeline\` first.
- Does **not** add a new public parameter. Removes the earlier \`min_non_nan_last_window\` approach.

Closes #1196

## Behavior
- \`dropna_last_window=False\`: unchanged
- \`dropna_last_window=True\` + NaN-tolerant estimator + \`differentiation is None\`: keep levels with NaNs in the last window
- \`dropna_last_window=True\` + non-tolerant estimator **or** differentiation set: \`dropna(how=\"any\")\` (current behavior)
- No extra warning (existing \`MissingValuesWarning\` in \`check_predict_input\` covers this)

## Test plan
- [x] Helper tests for lightgbm/catboost/xgboost/HistGradientBoosting (+ Pipeline unwrap)
- [x] Fold extraction: NaN-tolerant keeps level
- [x] Fold extraction: non-tolerant drops level
- [x] Fold extraction: differentiation forces drop even for NaN-tolerant estimators
- [x] Existing \`test_extract_data_folds_multiseries\` suite still passes

### CODE REVIEW

**Comment by JavierEscobarOrtiz** [comment]
Hello @vikas-prasad-cx, please see the new comment in https://github.com/skforecast/skforecast/issues/1196

**Comment by vikas-prasad-cx** [comment]
Thanks for the design clarification — I've rewritten this PR to match the intended approach:

1. Added \`estimator_has_native_nan_support\` in \`utils\` (module-based detection shared with categorical configuration; unwraps \`Pipeline\`).
2. Updated \`_extract_data_folds_multiseries\` so \`dropna_last_window=True\` keeps NaN levels only when the estimator supports NaNs **and** \`differentiation is None\`; otherwise keeps \`dropna(how=\"any\")\`.
3. Removed the \`min_non_nan_last_window\` public parameter entirely.
4. Added unit tests for NaN-tolerant / non-tolerant / differentiation / Pipeline cases.

Ready for another look when you have time.

**Comment by vikas-prasad-cx** [comment]
Rebased onto latest `0.24.x` and resolved the merge conflicts.

### What this addresses
Implements the design from [#1196](https://github.com/skforecast/skforecast/issues/1196) (maintainer comment by @JavierEscobarOrtiz):
- Added `estimator_has_native_nan_support` in `utils` (module-based family detection; unwraps `Pipeline`)
- When `dropna_last_window=True`, keep NaN last-window levels **only if** the estimator natively supports NaNs **and** `differentiation is None`
- Differentiation remains a hard exclusion (`dropna(how="any")`)
- No new public parameter / no extra warning

### Rebase notes
- Dropped the obsolete intermediate `min_non_nan_last_window` commit during rebase
- Replayed the rewrite commit cleanly on current `0.24.x` (preserving upstream `ForecasterFoundation` changes)

### Tests
```
pytest skforecast/utils/tests/tests_utils/test_estimator_has_native_nan_support.py \
       skforecast/model_selection/tests/tests_utils/test_extract_data_folds_multiseries.py
```
**24 passed**

Ready for another look when you have time.

**Comment by JavierEscobarOrtiz** [comment]
Hi @vikas-prasad-cx! Thanks for your contribution. We will review it in the coming weeks 😄
